README.md

```diff
@@ -1,8 +1,200 @@
+used by the `kubectl` command. When running inside a Kubernetes cluster it will
+use the configuration that Kubernetes mounts automatically in the pod file
+system. So in most cases this command line option won't have to be explicitly
+included.
+
+The `-alsologtostderr` option is very convenient when running the auto-heal
+service, both in development and production environments.
+
+Assuming that you want to have your own `my.yml` configuration file a typical
+command line will be the following:
+
+```bash
+$ autoheal -config-file my.yml -alsologtostderr
+```
+
+See the `autoheal.yml` file for a complete example.
+
+### AWX or AnsibleTower configuration
+
+The first section of the configuration file is named `awx` and it contains all
+the details needed to connect to the [AWX](https://www.ansible.com/products/awx-project)
+or [Ansible Tower](https://www.ansible.com/products/tower) server:
+
+```yaml
+awx:
+  address: https://myawx.example.com/api
+  proxy: http://myproxy.example.com:3128
+  credentialsRef:
+    namespace: my-namespace
+    name: my-awx-credentials
+  tlsRef:
+    namespace: my-namespace
+    name: my-awx-ca
+  project: "Auto-heal"
+```
+
+The `address` parameter is the URL of the API of the AWX server. It should
+contain the `/api` suffix, but not the `/v1` or `/v2` suffix, as the auto-heal
+service will internally decide which version to use.
+
+The `proxy` parameter is optional, and it indicates what HTTP proxy should be
+used to connect to the AWX API. If this parameter is not specified, or if it is
+empty, then the connection will be direct to the AWX server, without a proxy.
+
+The `credentialsRef` parameter is a reference to the [Kubernetes
+secret](https://kubernetes.io/docs/concepts/configuration/secret) that contains
+the user name and password used to connect to the AWX API. That secret should
+contain the `username` and `password` keys. For example:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: my-namespace
+  name: my-awx-credentials
+type: kubernetes.io/basic-auth
+data:
+  username: YWxlcnQtaGVhbGVy
+  password: ...
+```
+
+The `tlsRef` parameter is a reference to the [Kubernetes
+secret](https://kubernetes.io/docs/concepts/configuration/secret) that contains
+the certificates used to connect to the AWX API. That secret should contain at
+least the `ca.crt`, `tls.key` and `tks.crt` keys. For example:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: my-namespace
+  name: my-awx-tls
+type: kubernetes.io/tls
+data:
+  ca.crt: |-
+    LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUMvVENDQWVXZ0F3SUJBZ0lKQUxNRXB6OWxa
+    VkVzdzI3Sm5BYlMyejNhbUF0YTc1QmNnVGcvOUFCdDV0VVc2VTJOKzkKbXc9PQotLS0tLUVORCBD
+    ...
+  tls.key: ""
+  tls.crt: ""
+```
+
+The `ca.crt` key should contain the certificates of the authorities that are for
+verifying the certificate presented by the AWX server.
+
+The `tls.key` and `tls.crt` are only used when it is necessary to use a client
```

Comment:
Do we really want to support using client certificate for AWX auth?

Reply:
Probably not.

Reply:
In this case we should probably not mention it in the docs, and only add it if someone asks for it specifically :)

Reply:
I agree, will do.